command/agent/keyring_endpoint.go

```diff
@@ -0,0 +1,140 @@
+package agent
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// KeyringRequest is used route operator/raft API requests to the implementing
+// functions.
+func (s *HTTPServer) KeyringRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+
+	path := strings.TrimPrefix(req.URL.Path, "/v1/operator/keyring/")
+	switch {
+	case strings.HasPrefix(path, "keys"):
+		switch req.Method {
+		case http.MethodGet:
+			return s.keyringListRequest(resp, req)
+		case http.MethodPost, http.MethodPut:
```

Comment:
My only concern about doubling the ways to access the same endpoint is that it creates another API route that doesn't provide any end-user value, could actually create confusion, and prevents us from using the verb later should we want to.

Reply:
Yeah I felt conflicted about that. I'll remove the `/v1/operator/keyring/key/` version.